### ISSUE TITLE
`add_glow_effects(gradient_fill=True)`  treats step plot like a line plot

### ISSUE THREAD

**Issue by diceroll123** [dhaitz/mplcyberpunk#15]
```py
plt.style.use("cyberpunk")

ax = plt.gca()
ax.step(range(7), [0, 1, 2, 3, 2, 1, 0], marker="o")
mplcyberpunk.add_glow_effects(gradient_fill=True)
plt.show()
```

![image](https://user-images.githubusercontent.com/1243957/163696935-3c940089-a54f-4193-b668-9b26c0bea9a7.png)

This doesn't seem like an easy one.

**Comment by cedrichol** [comment]
I think the following works best, as it is using the auto-generated matplolib path instead of the data, it works out-of-the-box with every kind of plot ( 3 steps and the normal lines)
```python
        # xy = np.column_stack([x, y])
        # xy = np.vstack([[xmin, Ay], xy, [xmax, Ay], [xmin, Ay]])
        # clip_path = Polygon(xy, facecolor='none', edgecolor='none', closed=True)
        # ax.add_patch(clip_path)
        # im.set_clip_path(clip_path)

        transformedpath = line._get_transformed_path()
        path = line.get_path()
        extras = Path([[xmax,Ay],[xmin, Ay]], np.full(2, Path.MOVETO))
        extras.codes[:] = Path.LINETO
        path = path.make_compound_path(path, extras)
        im.set_clip_path(path, line._transform)
```

Here are the results:
![test_gradient_step](https://user-images.githubusercontent.com/65404079/184167329-892f4e0c-f966-4c44-a9a8-db06537e139c.png)

**Comment by diceroll123** [comment]
Ahhh you found what I was looking for, awesome!

**Comment by dhaitz** [comment]
Thanks to @cedrichol, this should be implemented with #20 and published in v0.6.0. @diceroll123 feel free to check!

**Comment by diceroll123** [comment]
Works great.
